### PR TITLE
[MRG+1]: Allow re-referencing to average reference

### PR DESCRIPTION
**Pull request by wmvanvliet**
There was a check in place that explicitly disallows moving from a custom reference to an average reference. Upon consideration, I cannot think of any reason why this should be disallowed. When cleaning EEG data (marking bad channels, etc.) it is quite common to first use a custom reference and when the data is clean, move to an average reference.

This PR removes the check and allows re-referencing to an average reference.

### CODE REVIEW

File: mne/io/proj.py [hunk 1]
```diff
@@ -669,11 +669,6 @@ def make_eeg_average_ref_proj(info, activate=True, verbose=None):
     eeg_proj: instance of Projection
         The SSP/PCA projector.
     """
-    if info.get('custom_ref_applied', False):
-        raise RuntimeError('Cannot add an average EEG reference projection '
-                           'since a custom reference has been applied to the '
-                           'data earlier.')
-
```
Comment (Eric89GXL) [inline mne/io/proj.py]: maybe change this to a `logger.info('Removing existing custom reference')`

Comment (Eric89GXL) [inline mne/io/proj.py]: and you need to set `info['custom_ref_applied'] = False` in this conditional right?

**Comment by Eric89GXL** [comment]
Otherwise LGTM

**Comment by agramfort** [comment]
merge when you're happy

**Comment by wmvanvliet** [comment]
This is better. Moving from one reference to another is allowed, but only by using the proper `io.set_eeg_reference` function, which assures that `custom_ref_applied` remains valid, even when things go wrong.

File: mne/io/tests/test_reference.py [hunk 2]
```diff
@@ -308,8 +316,6 @@ def test_add_reference():
     epochs = Epochs(raw, events=events, event_id=1, tmin=-0.2, tmax=0.5,
                     picks=picks_eeg, preload=True, proj='delayed')
     epochs_ref = add_reference_channels(epochs, 'Ref', copy=True)
-    # CAR after custom reference is an Error
-    assert_raises(RuntimeError, epochs_ref.set_eeg_reference)
```
Comment (Eric89GXL) [inline mne/io/tests/test_reference.py]: Can you add a test that after custom ref, you can go to average, and that before/after the info flag changes?

**Comment by agramfort** [comment]
LGTM

@Eric89GXL merge if you're happy

File: mne/io/reference.py [hunk 3]
```diff
@@ -287,7 +287,17 @@ def set_eeg_reference(inst, ref_channels=None, copy=True):
...
+            # Creating an average reference may fail. In this case, make sure
+            # that the custom_ref_applied flag is left untouched.
+            custom_ref_applied = inst.info['custom_ref_applied']
+            try:
+                inst.info['custom_ref_applied'] = False
+                inst.add_proj(make_eeg_average_ref_proj(inst.info,
+                              activate=False))
+            except Exception as e:
+                inst.info['custom_ref_applied'] = custom_ref_applied
+                raise e
```
Comment (Eric89GXL) [inline mne/io/reference.py]: Just do this:
```
except Exception:
    ...
    raise
```
It leads to a cleaner traceback

**Comment by Eric89GXL** [comment]
After one minor change LGTM

**Comment by Eric89GXL** [comment]
Thanks @wmvanvliet

**Pull request merged**